### DISCUSSION TITLE
Github Action Org level self-hosted runners didn't pick up jobs

### DISCUSSION THREAD

**Discussion by lawalong** [community/community#25361] [Actions]
<p>Hey Guys,</p>
<p>I got 2 runners on my K8S, and set them accessible for private repos.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/04ac14cc60ddc2cc59f081ddb53e21fe3c3f3bf3.png" data-download-href="/uploads/short-url/FkAIAKPwxeoHEZAPSn5jtVXsd5.png?dl=1" title="Selection_046" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/11696498/181090607-ecfa9b8f-171b-4d3d-a0f0-bec90f2bc9c7.png" alt="Selection_046" data-base62-sha1="FkAIAKPwxeoHEZAPSn5jtVXsd5" width="690" height="417" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/0/04ac14cc60ddc2cc59f081ddb53e21fe3c3f3bf3_2_690x417.png, https://aws1.discourse-cdn.com/github/optimized/2X/0/04ac14cc60ddc2cc59f081ddb53e21fe3c3f3bf3_2_1035x625.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/0/04ac14cc60ddc2cc59f081ddb53e21fe3c3f3bf3.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/04ac14cc60ddc2cc59f081ddb53e21fe3c3f3bf3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Selection_046</span><span class="informations">1063×643 74.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But I can see them in repos side, and of course queued jobs not been picked up.</p>
<p>Do I missing some setting?</p>
<p>Cheers,</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/lawalong">@lawalong</a>,</p>
<p>There are few points need you pay attention to:</p>
<ol>
<li>
<p>On the repository’s <strong>Settings</strong> &gt; <strong>Actions</strong> page,  only the self-hosted runners installed in current repository are listed.</p>
</li>
<li>
<p>If you also have installed a repo-level self-hosted runner in your repository, and this runner has the same labels with the org-level runner. When you use the same labels to specify self-hosted runner in the workflow, the workflow will only use the repo-level self-hosted runner.<br>
For example:<br>
– the org-level self-hosted runner with the labels “<strong>self-hosted, Linux, X64</strong>”<br>
– the repo-level self-hosted runner with the same labels “<strong>self-hosted, Linux, X64</strong>”<br>
– the workflow in the repository</p>
</li>
</ol>
<pre><code class="lang-yaml">jobs:
  job1:
    runs-on: [self-hosted, Linux, X64]
    steps:
    . . .
  job2:
    runs-on: [self-hosted, Linux, X64]
    steps:
    . . .
</code></pre>
<p><strong>job1</strong> and <strong>job2</strong> will only use the repo-level self-hosted runner.</p>
<ol start="3">
<li>You need to add some special labels to distinguish between repo-level and org-level self-hosted runners.<br>
For example:<br>
– the org-level self-hosted runner with the labels “<strong>self-hosted, Linux, X64, org-level</strong>”<br>
– the repo-level self-hosted runner with the labels “<strong>self-hosted, Linux, X64</strong>”<br>
– the workflow in the repository</li>
</ol>
<pre><code class="lang-yaml">jobs:
  job1:
    runs-on: [self-hosted, Linux, X64, org-level]
    steps:
    . . .
  job2:
    runs-on: [self-hosted, Linux, X64]
    steps:
    . . .
</code></pre>
<p><strong>job1</strong> will use the org-level self-hosted runner and <strong>job2</strong> will use the repo-level self-hosted runner.</p>

**Comment by lawalong** [comment]
<p><a class="mention" href="/brightran">@brightran</a>  Thanks for your advise about label</p>

**Comment by lawalong** [comment, marked as answer]
<p>Problem solved.</p>
<p>Runner version 2.263.0 can correctly pick up job</p>
<p>I was on 2.262.1, it will queue forever</p>
